### DISCUSSION TITLE
aioboto3 type hints

### DISCUSSION THREAD

**Discussion by markonosetypless** [youtype/mypy_boto3_builder#214] [Q&A] [answered]
Hello, I have a question.

I don't know if I'm using this wrong but for example:

```
    import aioboto3

    awaitable_resource = aioboto3.Session().resource("dynamodb")
    resource = await awaitable_resource.__aenter__()

    ...

    resource.__aexit__(None, None, None)
```

PyCharm highlights the ```__aenter__()``` method and shows  ```Unresolved attribute reference '__aenter__' for class 'DynamoDBServiceResource' ``` but no problem for the ```__aexit__```

And then when using

```
        table = await resource.Table("table_name")
        response = await table.query(...)
        item = response["Items"][0]
```

the type of item is 

```
AttributeValueTableTypeDef = TypedDict(
    "AttributeValueTableTypeDef",
    {
        "S": str,
        "N": str,
        "B": bytes,
        "SS": List[str],
        "NS": List[str],
        "BS": List[bytes],
        "M": Dict[str, Dict[str, Any]],
        "L": List[Dict[str, Any]],
        "NULL": bool,
        "BOOL": bool,
    },
)
```
But the actual type of the item should just be a dictionary with the Union of the types defined in the AttributeValueTableTypeDef dictionary.

So to me it seems that by installing ```types-aioboto3``` you only get the types of ```types-aiobotocore```.

What am I doing wrong?

**Comment by vemel** [comment, marked as answer]
Hello!

It was fixed in the newest version, but I did not build aio libs with the new version yet. I am going to do it today.

**Reply by markonosetypless** [reply]
This was an amazingly fast reply. Thanks for the info!

**Reply by vemel** [reply]
No so fast release though :) Feel free to update to `types-aiobotocore 2.5.2.post1` and let me know if it works for you as expected.
